src/Components/test/E2ETest/ServerExecutionTests/InteropReliabilityTests.cs

```diff
@@ -258,31 +215,31 @@ public async Task CannotInvokeJSInvokableMethodsWrongReferenceIdType()
                 JsonSerializer.Serialize(new object[] { new { __dotNetObject = 1 } }));
 
             // Assert
-            Assert.Single(DotNetCompletions, c => c.Message == expectedError);
+            Assert.Single(DotNetCompletions, c => c == expectedError);
             await ValidateClientKeepsWorking(Client, Batches);
         }
 
         [Fact]
-        [Flaky("https://github.com/aspnet/AspNetCore/issues/13086", FlakyOn.AzP.Windows)]
         public async Task ContinuesWorkingAfterInvalidAsyncReturnCallback()
         {
             // Arrange
             var expectedError = "An exception occurred executing JS interop: The JSON value could not be converted to System.Int32. Path: $ | LineNumber: 0 | BytePositionInLine: 3.. See InnerException for more details.";
 
-            await GoToTestComponent(Batches);
-
             // Act
             await Client.ClickAsync("triggerjsinterop-malformed");
 
             var call = JSInteropCalls.FirstOrDefault(call => call.Identifier == "sendMalformedCallbackReturn");
             Assert.NotEqual(default, call);
 
             var id = call.AsyncHandle;
-            await Client.HubConnection.InvokeAsync(
-                "EndInvokeJSFromDotNet",
-                id,
-                true,
-                $"[{id}, true, \"{{\"]");
+            await Client.ExpectRenderBatch(async () =>
+            {
+                await Client.HubConnection.InvokeAsync(
```

Comment:
nit: No need for await here

Reply:
I'm not going to clean all of these up - these tests in general have a fix. Write a code-fix if you're deeply passionate about this :)

Reply:
That's why I said nit 😜. Also, we should never use InternalsVisibleTo ever 😄